src/main/scala/aserralle/akka/stream/kcl/scaladsl/KinesisWorkerSource.scala

```diff
@@ -40,8 +42,14 @@ object KinesisWorkerSource {
           val worker = workerBuilder(
             new IRecordProcessorFactory {
               override def createProcessor(): IRecordProcessor =
-                new IRecordProcessor(queue.offer,
-                                     settings.terminateStreamGracePeriod)
+                new IRecordProcessor(
+                  { record =>
+                    //not sure about infinite await here, probably need to externalize it,
+                    // user mast provide grace period in config or something
+                    Await.result(queue.offer(record), Duration.Inf)
+                  },
```

Comment:
Can you make it configurable? Just to make it more visible. "BackpressureTimeout" would be a fine name for that.

Also, please remove the curly brackets and the comment.